http/stateful-basic/src/main/java/org/wildfly/security/http/sfbasic/BasicAuthenticationMechanism.java

```diff
@@ -196,7 +215,53 @@ public void evaluateRequest(final HttpServerRequest request) throws HttpAuthenti
         request.noAuthenticationInProgress(response -> prepareResponse(request, displayRealmName, response));
     }
 
+    private boolean attemptReAuthentication(final HttpServerRequest request) throws HttpAuthenticationException {
+        String sessionID = null;
+        for (HttpServerCookie cookie : request.getCookies()) {
+           if (COOKIE_NAME.equals(cookie.getName())) {
+               sessionID = cookie.getValue();
+               break;
+           }
+        }
+
+        if (sessionID != null) {
+            BasicIdentityCache identityCache = new BasicIdentityCache(sessionID);
+            CachedIdentityAuthorizeCallback authorizeCallback = new CachedIdentityAuthorizeCallback(identityCache);
+            try {
+                callbackHandler.handle(new Callback[]{authorizeCallback});
+            } catch (IOException | UnsupportedCallbackException e) {
+                throw new HttpAuthenticationException(e);
+            }
+            if (authorizeCallback.isAuthorized()) {
+                try {
+                    succeed();
+                } catch (IOException | UnsupportedCallbackException e) {
+                    throw new HttpAuthenticationException(e);
+                }
+                // We need the cookie to be sent after re-authentication to extend the validity.
+                request.authenticationComplete(identityCache, identityCache::remove);
+                request.resumeRequest();
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private boolean authorize(String username, IdentityCache identityCache)
+            throws HttpAuthenticationException {
+        CachedIdentityAuthorizeCallback authorizeCallback = new CachedIdentityAuthorizeCallback(username, identityCache);
+        try {
+            callbackHandler.handle(new Callback[] { authorizeCallback });
+            return authorizeCallback.isAuthorized();
+        } catch (IOException | UnsupportedCallbackException e) {
+            throw new HttpAuthenticationException(e);
+        }
+    }
+
     private void prepareResponse(final HttpServerRequest request, String realmName, HttpServerResponse response) {
+        System.out.println("We are entering challenging times.");
```

Comment:
This print statement should be removed or changed to a trace message.

Reply:
Ah sorry, was reviewing the individual commit and realized that this was removed later on. You can ignore this. :)